sdcm/nemesis.py

```diff
@@ -4413,6 +4419,77 @@ def disrupt_add_remove_mv(self):
         finally:
             self.unset_current_running_nemesis(cql_query_executor_node)
 
+    def disrupt_toggle_audit(self):
+        """
+            Enable audit log with all categories and user keyspaces (if audit already enabled, disable it and finish the Nemesis),
+            verify audit log content,
+            reduce categories by excluding DML and QUERY,
+            verify DDL are logged in audit log correctly. Leaves audit log enabled this way.
+        """
+        if not self.target_node.is_enterprise:
+            raise UnsupportedNemesis("Auditing feature is only supported by Scylla Enterprise")
+
+        if not (test_ks_cf_list := self.cluster.get_non_system_ks_cf_list(db_node=self.target_node)):
+            raise UnsupportedNemesis("Not found non-system keyspaces")
+
+        audit = Audit(self.cluster)
+
+        if audit.is_enabled():
+            InfoEvent("Audit was enabled -> disabling it").publish()
+            audit.disable()
+            return
+
+        keyspaces_for_audit = [ks_cf.split(".")[0] for ks_cf in test_ks_cf_list]
+        InfoEvent("Enabling full audit").publish()
+        audit_config = AuditConfiguration(
+            store="table",
+            categories=["DCL", "DDL", "AUTH", "ADMIN", "DML", "QUERY"],
```

Comment:
I'm think we should make sure this nemesis ends, we don't leave audit with QUERY, it would break most of the test cases...

Reply:
Not sure what it will cause, we tested it in the past and it reduced the throughput significantly.